File: lib/models/auth_state.dart
class AuthState {
  const AuthState({
    required this.baseUrl,
    required this.deviceId,
    required this.clientName,
    required this.deviceName,
    required this.appVersion,
    required this.userId,
    required this.accessToken,
    this.disableTls = false,
  });

  final String baseUrl;
  final String deviceId;
  final String clientName;
  final String deviceName;
  final String appVersion;
  final String userId;
  final String accessToken;
  final bool disableTls;

  bool get isLoggedIn => accessToken.isNotEmpty && userId.isNotEmpty;
}

File: app/src/main/java/com/muufin/compose/model/AuthState.kt
package com.muufin.compose.model


data class AuthState(
    val baseUrl: String = "",
    val userId: String = "",
    val accessToken: String = "",
    val deviceId: String = "",
    val clientName: String = "Muufin",
    val deviceName: String = android.os.Build.MODEL ?: "Android",
    val appVersion: String = "0.1.0",
    val disableTls: Boolean = false,
    
    val customCaBase64: String = "",
) {
    val isSignedIn: Boolean get() = baseUrl.isNotBlank() && userId.isNotBlank() && accessToken.isNotBlank()
}
